### ISSUE TITLE
invariant_booleans false positive

### ISSUE THREAD

**Issue by xavierhainaux** [dart-lang/linter#332]
```dart
main() {
  int a = new DateTime.now().millisecondsSinceEpoch;

  innerFunction() {
    if (a > 0) {
      return false;
    }
  }

  if (a > 0) { // Lint reported here :-(
    print('bla');
  }
}
```

The lint seems to think that the `innerFunction` is part of the flow.

Dart 1.20.1
